### PR TITLE
1219 Enhance SplitChannel transform

### LINKED ISSUE #1219: SplitChannels transform
**Is your feature request related to a problem? Please describe.**
a splitchannel transform is needed in order to preprocess each subset of channels separately.

**Describe alternatives you've considered**
the current implementation only works as a post-processing module

**Additional context**
https://github.com/Project-MONAI/MONAI/blob/4e9164032610b69acdf716765909f5ec393af36a/monai/transforms/post/array.py#L29-L34

### PR DESCRIPTION
**Pull request by Nic-Ma**
Fixes #1219 .

### Description
This PR enhanced `SplitChannel` and `SplitChanneld` transforms:
1. Removed `to_onehot` logic.
2. Added support for numpy data.
3. Added support to set channel dim.

### Status
**Ready**

### Types of changes

- [x] Non-breaking change (fix or new feature that would not break existing functionality).
- [ ] Breaking change (fix or new feature that would cause existing functionality to change).
- [ ] New tests added to cover the changes.
- [ ] Integration tests passed locally by running `./runtests.sh --codeformat --coverage`.
- [ ] Quick tests passed locally by running `./runtests.sh --quick`.
- [ ] In-line docstrings updated.
- [ ] Documentation updated, tested `make html` command in the `docs/` folder.

### CODE REVIEW

File: monai/transforms/utility/dictionary.py [hunk 1]
```diff
@@ -154,6 +155,48 @@ def __call__(self, data: Mapping[Hashable, np.ndarray]) -> Dict[Hashable, np.nda
...
+                for example: if the key of input data is `pred` and split 2 classes, the output
+                data keys will be: pred_(output_postfixes[0]), pred_(output_postfixes[1])
+            channel_dim: which dimension of input image is the channel, default to None
+                to automatically select: if data is numpy array, channel_dim is 0 as
+                `numpy array` is used in the pre transforms, if PyTorch Tensor, channel_dim
+                is 1 as in most of the cases `Tensor` is uses in the post transforms.
+
+        """
+        super().__init__(keys)
+        self.output_postfixes = output_postfixes
```
Comment (wyli) [inline monai/transforms/utility/dictionary.py:185]: could you add a default `output_postfixes`, to `keys_0`, `keys_1`, ... raise an error if any of the keys already exist  in `data`

**Review by wyli — APPROVED** [review]

File: monai/transforms/utility/dictionary.py (hunk 1 above)
Comment (Nic-Ma) [reply monai/transforms/utility/dictionary.py:186]: OK, updated in the latest commit, could you please help review it again?

Thanks.

**Pull request merged**